### PR TITLE
Allow updating draft releases while keeping them as draft

### LINKED ISSUE #162: Can't save Release draft as draft
- Gitea version (or commit ref): 0.9.99.0915
- Git version: 2.10
- Operating system: CentOS 64bit
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] SQLite
- Log gist:

## Description

Can't save a Release draft modification as draft.

## How to reproduce

1. Go to release tab
2. Create a new draft and click on "Save Draft"
3. On release tab, click on "Edit" button for the newly created draft
4. The "Save Draft" button not appearing

### PR DESCRIPTION
**Pull request by strk**
Closes #162

### CODE REVIEW

**Comment by 0xbaadf00d** [comment]
LGTM

File: routers/repo/release.go [hunk 1]
```diff
@@ -247,6 +247,7 @@ func EditRelease(ctx *context.Context) {
 	ctx.Data["title"] = rel.Title
 	ctx.Data["content"] = rel.Note
 	ctx.Data["prerelease"] = rel.IsPrerelease
+	ctx.Data["draft"] = rel.IsDraft
```
Comment (bkcsoft) [inline routers/repo/release.go]: Should be `ctx.Data["IsDraft"]`

If you feel like it, you might as well fix the above ones as well. Though not related to this issue :slightly_smiling_face:

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: routers/repo/release.go (hunk 1 above)
Comment (strk) [reply routers/repo/release.go]: I'd prefer keeping commits focused on what they address. There are also still ALL_UPPER_CASE constants in that file, but would rather not touch now. Can change the IsDraft though, being a new one.

**Review by bkcsoft — APPROVED** [review]

**Comment by bkcsoft** [comment]
LGTM

**Comment by thibaultmeyer** [comment]
LGTM

**Pull request merged**